src/hosted-buttons/utils.js

```diff
@@ -90,14 +93,22 @@ export const getHostedButtonDetails: HostedButtonDetailsParams = async ({
 
   // $FlowIssue request returns ZalgoPromise
   const { body } = response;
-  const variables = body.button_details.link_variables;
+  const { link_variables: variables, preferences } = body.button_details;
+
   return {
     style: {
       layout: getButtonVariable(variables, "layout"),
       shape: getButtonVariable(variables, "shape"),
       color: getButtonVariable(variables, "color"),
       label: getButtonVariable(variables, "button_text"),
+      height: getButtonVariable(variables, "height"),
+    },
+    version: body.version,
+    preferences: {
+      secondButton: preferences?.second_button,
+      eligibleFundingMethods: preferences?.eligible_funding_methods,
```

Comment:
I'm curious about conditionally setting the `preferences` property, like:

```js
...(preferences && {
  preferences: {
    secondButton: preferences.second_button,
    eligibleFundingMethods: preferences.eligible_funding_methods,
  }
})
```

so that trying to reference `preferences.secondButton` accidentally (in a v1 context) will error instead of a quieter `undefined`. I only had this thought because when I was reading the tests for `v1` I saw a reference to `preferences` and it confused me at first.

I like what you have as-is ( :shipit: )  but am curious what you think about this!

Reply:
This is a really good solution if we don't want to return `preferences` if it doesn't exist!  I also didn't think about this syntax but its really cool. 😯

If we didn't include `preferences` here, we would probably have to check anyway for it anytime we use it downstream.  So my thoughts were to just always include it in the returned object and destructure it downstream. 

I included it in the tests for `v1` for this use case, but I could remove it if we didn't want to return `preferences` everytime this is called 😄